src/storage.rs

```diff
@@ -115,9 +123,9 @@ impl From<EvmCode> for Bytecode {
                     raw: raw.into(),
                 })
             }
-            EvmCode::Legacy(code) => unsafe {
-                Bytecode::new_analyzed(code.bytecode, code.original_len, JumpTable(code.jump_table))
-            },
+            EvmCode::Eof(eof_code) => Eof::decode(eof_code.0.clone())
+                .map(|eof| Bytecode::Eof(Arc::new(eof)))
+                .unwrap_or_else(|_| Bytecode::Eof(Arc::new(Eof::default()))),
```

Comment:
I'm not a fan of implicit corner logic like this. Let's panic with a `// TODO: Rewrite as [TryFrom]` added before `impl From` (also covers the `unsafe` in the `Legacy` arm).

Reply:
@byhashdong Last nit but this should be an `unwrap` now :pray:.

Reply:
Ok, Got it!